tests/scipy_stats_test.py

```diff
@@ -80,6 +80,42 @@ def args_maker():
                             tol=1e-3)
       self._CompileAndCheck(lax_fun, args_maker)
 
+  @genNamedParametersNArgs(2)
+  def testWrappedCauchyPdf(self, shapes, dtypes):
+    rng = jtu.rand_default(self.rng())
+    rng_uniform = jtu.rand_uniform(self.rng(), low=1e-3, high=1 - 1e-3)
+    scipy_fun = osp_stats.wrapcauchy.pdf
+    lax_fun = lsp_stats.wrapcauchy.pdf
+
+    def args_maker():
+      x = rng(shapes[0], dtypes[0])
+      c = rng_uniform(shapes[1], dtypes[1])
+      return [x, c]
+
+    with jtu.strict_promotion_if_dtypes_match(dtypes):
+      self._CheckAgainstNumpy(scipy_fun, lax_fun, args_maker, check_dtypes=False,
+                              tol=1e-3)
+      self._CompileAndCheck(lax_fun, args_maker, tol={np.float32: 1e-5,
+                                                      np.float64: 1e-11})
+
+  @genNamedParametersNArgs(2)
+  def testWrappedCauchyLogPdf(self, shapes, dtypes):
+    rng = jtu.rand_default(self.rng())
+    rng_uniform = jtu.rand_uniform(self.rng(), low=1e-3, high=1 - 1e-3)
+    scipy_fun = osp_stats.wrapcauchy.logpdf
+    lax_fun = lsp_stats.wrapcauchy.logpdf
+
+    def args_maker():
+      x = rng(shapes[0], dtypes[0])
+      c = rng_uniform(shapes[1], dtypes[1])
+      return [x, c]
+
+    with jtu.strict_promotion_if_dtypes_match(dtypes):
+      self._CheckAgainstNumpy(scipy_fun, lax_fun, args_maker, check_dtypes=False,
+                              tol=1e-3)
```

Comment:
Is there any reason you set so loose a tolerance here? I thought this previously was `1e-5` or `1e-13` depending on the dtype?

Reply:
Also, it looks like TPU needs a tolerance of 1E-4 for float32

Reply:
Apologies, I didn't realize the tolerance had default values when unspecified. I set `tol={np.float32: 1e-4}` in both pdf's and logpdf's `_CheckAgainstNumpy` to address the TPU error.

Since I do not own a TPU on my machine, is there any way to virtualize the operation and check the behavior for myself? If not, how do I understand which tolerance to set autonomously? Should I setup my dev environment on Colab TPU? Or is it possible to observe results of TPU test runs through the CI somehow?

Thank you for your time!

Reply:
Oh ok, I see this kind of back and forth is common ([ref](https://jax.readthedocs.io/en/latest/contributing.html#restricted-test-suite))

Reply:
Thanks - sorry about the churn here.